### DISCUSSION TITLE
Live Inference using YoloV8

### DISCUSSION THREAD

**Discussion by PiyushKamboj1503** [roboflow/supervision#78] [Q&A] [answered]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

The argument show=True give the live inference in YoloV8 with it own bounding boxes. 
How to get the live prediction in a window in similar way using supervision with the custom bounding boxes and zones.

**Comment by SkalskiP** [comment]
Hi 👋🏻! Given the fact that it is not an issue related to the library itself, I'll move that conversation to our Q&A discussions

**Comment by SkalskiP** [comment, marked as answer]
1. Use [get_video_frames_generator](https://roboflow.github.io/supervision/video/#get_video_frames_generator)
2. Run detection with YOLOv8 on the current frame
3. [Convert](get_video_frames_generator) YOLOv8 result to `Detections` object
4. Annotate the frame with [BoxAnnotator](https://roboflow.github.io/supervision/detection/annotate/#boxannotator).
5. Use OpenCV `imshow` to display annotated images.

Here is a pseudo-code possibly would require some fixing but could be useful as the first iteration.

```python
import cv2
import supervision as sv
from ultralytics import YOLO

model = YOLO('yolov8s.pt')
annotator = BoxAnnotator()

for frame in sv.get_video_frames_generator(source_path='source_video.mp4'):
    yolov8_results = model(frame)[0]
    detections = sv.Detections.from_yolov8(yolov8_results)
    annotated_frame = annotator.annotate(scene=frame.copy(), detections=detections)
    cv2.imshow('image window', annotated_frame)
```
